examples/html-preview/pages/[boardId]/[path].tsx

```diff
@@ -0,0 +1,265 @@
+import { GetServerSideProps } from "next";
+import initMiro from "../../initMiro";
+
+import { WidgetItem } from "@mirohq/miro-api/dist/highlevel/Item";
+import { Board, FrameItem, ShapeItem } from "@mirohq/miro-api";
+import { FC, PropsWithChildren } from "react";
+
+interface Item {
+  id: number;
+  type: string;
+  content: string;
+  image: string;
+  style: ShapeItem["style"];
+  geometry: WidgetItem["geometry"];
+  position: WidgetItem["position"];
+}
+
+export const getServerSideProps: GetServerSideProps = async ({
+  req,
+  res,
+  query,
+}) => {
+  const { miro } = initMiro(req);
+
+  // cache for 10 min
+  res.setHeader(
+    "Cache-Control",
+    "public, s-maxage=600, stale-while-revalidate=59"
+  );
+
+  // redirect to auth url if user has not authorized the app
+  if (!(await miro.isAuthorized(""))) {
+    return {
+      redirect: {
+        destination: miro.getAuthUrl(),
+        permanent: false,
+      },
+    };
+  }
+
+  const api = miro.as("");
+
+  const items: Item[] = [];
+
+  if (!query.boardId) {
+    return {
+      notFound: true,
+    };
+  }
+
+  let board: Board;
+  try {
+    board = await api.getBoard(query.boardId.toString());
+  } catch (err) {
+    return { notFound: true };
+  }
+
+  const frames: FrameItem[] = [];
+  for await (const frame of board.getAllItems({ type: "frame" })) {
+    if (frame instanceof FrameItem) frames.push(frame);
+  }
+
+  const path = query.path || "index";
+
+  const frame =
+    frames.filter((frame) => frame.data?.title == path)[0] ||
+    frames[0] ||
+    board;
+
+  for await (let item of frame.getAllItems()) {
```

Comment:
This file is a bit long. Can we extract some functionality? `const board = getBoard(id); const frame = getFrame(frameId)` etc

Reply:
https://github.com/mettin/next-server-test/tree/main/src/utils/fetch

Reply:
I've tried to split it now in separate components. With server side data fetching in one place and frontend components in the other.